addons/account/models/account_account.py

```diff
@@ -195,15 +195,13 @@ def _check_used_as_journal_default_debit_credit_account(self):
             return
 
         self.flush(['reconcile'])
-        self._cr.execute('''
+        queries = [f'''
             SELECT journal.id
             FROM account_journal journal
-            WHERE journal.payment_credit_account_id in %(credit_account)s
-            OR journal.payment_debit_account_id in %(debit_account)s ;
-        ''', {
-            'credit_account': tuple(accounts.ids),
-            'debit_account': tuple(accounts.ids)
-        })
+            WHERE journal.payment_{x}_account_id IN %(accounts)s
+            AND journal.default_account_id != journal.payment_{x}_account_id
+        ''' for x in ('debit', 'credit')]
+        self._cr.execute(' UNION ALL '.join(queries), {'accounts': tuple(accounts.ids)})
```

Comment:
Shouldn't you de-duplicate the ids somewhere? (or maybe keep the structure of the old query?)